src/main/scala/strawman/collection/Set.scala

```diff
@@ -76,38 +76,7 @@ trait SetMonoTransforms[A, +Repr]
 
 }
 
-trait SetPolyTransforms[A, +C[X]] extends IterablePolyTransforms[A, C] {
-
-  /** Creates a new $coll by adding all elements contained in another collection to this $coll, omitting duplicates.
-    *
-    * This method takes a collection of elements and adds all elements, omitting duplicates, into $coll.
-    *
-    * Example:
-    *  {{{
-    *    scala> val a = Set(1, 2) concat Set(2, 3)
-    *    a: scala.collection.immutable.Set[Int] = Set(1, 2, 3)
-    *  }}}
-    *
-    *  @param that     the collection containing the elements to add.
-    *  @return a new $coll with the given elements added, omitting duplicates.
-    */
-  def concat(that: Set[A]): C[A]
-
-  /** Alias for `concat` */
-  @inline final def ++ (that: Set[A]): C[A] = concat(that)
-
-  /** Computes the union between of set and another set.
-    *
-    *  @param   that  the set to form the union with.
-    *  @return  a new set consisting of all elements that are in this
-    *  set or in the given set `that`.
-    */
-  @inline final def union(that: Set[A]): C[A] = concat(that)
-
-  /** Alias for `union` */
-  @inline final def | (that: Set[A]): C[A] = concat(that)
-
-}
+trait SetPolyTransforms[A, +C[X]] extends IterablePolyTransforms[A, C]
```

Comment:
I moved the `concat` operation into the `immutable` hierarchy. I think transformation operations (those defined in `XxxTransforms` traits) should all go to the immutable hierarchy.

If we kept these operations that would be a little bit confusing to have both `++` and `++=`, `map` and `mapInPlace`, etc.

I think the typical use case of mutable collections is to mutate the collection in place. Therefore this usage should be the most straightforward.

If one wants to copy a mutable collection into a new one, he should have to explicitly tell that, because that might be an mistake. He can do it like so:

~~~ scala
val bufferCopy = buffer.view.map(f).to(ListBuffer)
~~~

Here, the view makes it possible to apply transformation operations to the mutable collection without having to first transform it into an immutable collection.